### PR TITLE
Fix #919 - support standard introspection

### LINKED ISSUE #919: Support standard introspection for oauth2 token validation
### Describe your problem(s)
MGW at the moment supports connecting with WSO2 APIM in order to validate opaque oauth2 tokens. It communicates to the key validation service of the WSO2 APIM.
We need to support validating standard oauth2 opaque token using the standard introspection endpoint.
### Describe your solution
Introduce a new security handler in order to validate with standard introspect endpoint.

### PR DESCRIPTION
**Pull request by Rajith90**
### Purpose
Support for validating opaque oauth2 tokens using the standard introspection.

### Issues
Fixes #919

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 1]
```diff
@@ -112,29 +114,58 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
     };
-    OAuth2KeyValidationProvider oauth2KeyValidationProvider = new (keyValidationServerConfig);
+    oauth2:IntrospectionServerConfig  keyValidationConfig = {
+        url: getConfigValue(KM_CONF_INSTANCE_ID, KM_SERVER_URL, "https://localhost:9443"),
+        clientConfig: clientConfig
+    };
+    string introspectURL = getConfigValue(KM_CONF_INSTANCE_ID, KM_SERVER_URL, "https://localhost:9443");
+    string keymanagerContext = getConfigValue(KM_CONF_INSTANCE_ID, KM_TOKEN_CONTEXT, "oauth2");
+    introspectURL = (introspectURL.endsWith(PATH_SEPERATOR)) ? introspectURL + keymanagerContext : introspectURL + PATH_SEPERATOR + keymanagerContext;
+    introspectURL = (introspectURL.endsWith(PATH_SEPERATOR)) ? introspectURL + INTROSPECT_CONTEXT : introspectURL + PATH_SEPERATOR + INTROSPECT_CONTEXT;
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: `introspectURL` is going to be the result of 157 anyway. Result of 156 is not used. Mistake?

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 2]
```diff
@@ -187,3 +218,104 @@ function initiateKeyManagerConfigurations() {
...
+    return scopesArray;
+}
+
+function getCredentialBearer() returns http:CredentialBearer {
+    string crednetailBearerString= getConfigValue(KM_CONF_SECURITY_OAUTH2_INSTANCE_ID, CREDENTIAL_BEARER, http:AUTH_HEADER_BEARER);
+    if (stringutils:equalsIgnoreCase(crednetailBearerString, http:AUTH_HEADER_BEARER)) {
+        return http:AUTH_HEADER_BEARER;
+    } else if (stringutils:equalsIgnoreCase(crednetailBearerString, http:POST_BODY_BEARER)) {
+        return http:POST_BODY_BEARER;
+    } else {
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: else statement is not required. `return http:NO_BEARER;` will do the job.

File: distribution/resources/conf/micro-gw.conf [hunk 3]
```diff
@@ -14,10 +14,40 @@ removeAuthHeaderFromOutMessage=true
...
+      enabled = false
+      accessToken = ""
+    [keymanager.security.oauth2.refresh]
+      enabled = false
+      refreshUrl = ""
+      scopes = ""
+      refreshToken = ""
+      clientId = ""
+      clientSecret = ""
+
```
Comment (praminda) [inline distribution/resources/conf/micro-gw.conf]:
```suggestion
```

File: distribution/resources/conf/micro-gw.conf [hunk 4]
```diff
@@ -14,10 +14,40 @@ removeAuthHeaderFromOutMessage=true
...
```
Comment (praminda) [inline distribution/resources/conf/micro-gw.conf]:
```suggestion
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal (hunk 1 above)
Comment (Rajith90) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Result of 156 is used in line 157

Comment (praminda) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Oops my mistake

**Pull request merged**